### DISCUSSION TITLE
Consulta sobre ejercicio de ejemplo en el TP10

### DISCUSSION THREAD

**Discussion by sanvaltos** [INGCOM-UNRN-P1/cursada-2023#226] [Preguntas y respuestas] [answered]
Hola, buenas tardes. Tengo una duda con el calculo de complejidad en el ejemplo dado en el TP10. 

#### Complejidad temporal:
El algoritmo tiene un único lazo, que se hace ‘divisor’ veces.

El costo computacional queda como:

1 + 1 + 7 + 7 + 1 + 1 + 1 + (dividendo/divisor) + 1 + 1

En donde dice '(dividendo/divisor)' corresponde a esta parte del código? pregunto porque es acá donde se hace la división lenta

```C
while (dividendo >= divisor)
        {
            dividendo -= divisor;
            cociente++;
        }
```

Si es así, ¿se puede escribir también como '... + (divisor) * (2 + 2) +... ' ? ya que las repeticiones del bucle están definidas en función de lo que valga dividendo y divisor.

**Comment by martinvilu** [comment, marked as answer]
> ya que las repeticiones del bucle están definidas en función de lo que valga dividendo y divisor.

Sí
